.travis.yml

```diff
@@ -0,0 +1,29 @@
+branches:
```

Comment:
While this means that the only branch that builds is main, you currently appear to have pull requests triggering builds (which is good for testing!) but this does mean you appear to be deploying on every PR

Instead, I suggest you use the [deploy step with the script provider](https://docs.travis-ci.com/user/deployment/script/) which should let you configure the deployment to only occur for a specific branch whilst allowing PRs to still trigger the script stage to run tests